src/shared/App.js

```diff
@@ -40,40 +40,29 @@ function App() {
       ) : (
         <Router>
           <Switch>
-            <Route exact path="/login">
-              {user ? <Redirect to="/" /> : <LoginRegister />}
-            </Route>
-            <Route exact path="/chakraExpoDashboard">
-              <ChakraProvider theme={dashboardTheme}>
-                <ChakraExpoDashboard />
-              </ChakraProvider>
-            </Route>
-            <Route exact path="/chakraExpoStore">
-              <ChakraProvider theme={storeTheme}>
-                <ChakraExpoStore />
-              </ChakraProvider>
-            </Route>
-            <Route exact path="/store">
-              <Store />
-            </Route>
+            <Route exact path="/login" render={() => (user ? <Redirect to="/" /> : <LoginRegister />)} />
+            <Route exact path="/chakraExpoDashboard" component={ChakraExpoDashboard} />
+            <Route
+              exact
+              path="/chakraExpoStore"
+              render={() => (
+                <ChakraProvider theme={storeTheme}>
+                  <ChakraExpoStore />
+                </ChakraProvider>
+              )}
+            />
+            <Route exact path="/store" component={Store} />
             <ProtectedRoute path="/">
               <Grid templateColumns={`${NAVBAR_WIDTH} 1fr`} h="100vh">
                 <Box borderRight="2px solid black" w="100%" h="100%">
                   {/** Navbar width is set manually to keep the position fixed */}
                   <Navbar w={NAVBAR_WIDTH} />
                 </Box>
-
                 <Box w="100%" h="100%" p="72px">
                   <Switch>
-                    <ProtectedRoute exact path="/home">
-                      <Dashboard />
-                    </ProtectedRoute>
-                    <ProtectedRoute exact path="/leaderboard">
-                      Leaderboard page, not yet implemented.
-                    </ProtectedRoute>
-                    <ProtectedRoute exact path="/team">
-                      <TeamView />
-                    </ProtectedRoute>
+                    <ProtectedRoute exact path="/home" component={Dashboard} />
+                    <ProtectedRoute exact path="/leaderboard" component={'Leaderboard page, not yet implemented.'} />
+                    <ProtectedRoute exact path="/team" component={TeamView} />
```

Comment:
heh I actually prefer the other method because it makes it read like a normal React component tree, but I can also see how this makes the code more concise